googledataprocauthenticator/google.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from sparkmagic.auth.customauth import Authenticator
+import json
+import os
+import urllib3.util
+import subprocess
+from sparkmagic.livyclientlib.exceptions import BadUserConfigurationException
+from sparkmagic.utils.constants import WIDGET_WIDTH
+from google.cloud import dataproc_v1beta2
+import google.auth.transport.requests 
+from google.auth import _cloud_sdk  
+from google.auth.exceptions import UserAccessTokenError
+from hdijupyterutils.ipywidgetfactory import IpyWidgetFactory
+from google.oauth2.credentials import Credentials
+
+
+# The name of the Cloud SDK shell script
+_CLOUD_SDK_POSIX_COMMAND = "gcloud"
+_CLOUD_SDK_WINDOWS_COMMAND = "gcloud.cmd"
+# The command to get all credentialed accounts 
+_CLOUD_SDK_USER_CREDENTIALED_ACCOUNTS_COMMAND = ("auth", "list", "--format", "json")
+
+def load_json_input(result):
```

Comment:
This whole function seems equivalent to just calling json.loads, since any exceptions are reraised. Is there are advantage to splitting this out?

Reply:
I think I had a reason when I was depending on it returning None for some if check later in the code, but its gone now so there's no longer any reason why this is helpful. I removed it.